opentracing-impl/src/main/java/io/opentracing/AbstractSpanBuilder.java

```diff
@@ -52,11 +52,17 @@ public final AbstractSpanBuilder addReference(String referenceType, SpanContext
 
     @Override
     public final AbstractSpanBuilder asChildOf(SpanContext parent) {
+        if (parent instanceof NoopSpanContext) {
+            return AbstractNoopSpanBuilder.INSTANCE;
```

Comment:
Implementations might want to return their own version of this.
I'm not sure the best way of achieving this.
Maybe the simplest is to remove the `final`  modifier, that way implementations can do it their way.

Keep in mind this is only a problem if a NoopSpanContext is actually passed in, and the implementation itself largely dictates whether that is possible.